lib/src/cwrapper.cc

```diff
@@ -0,0 +1,59 @@
+#include <facter/cwrapper.hpp>
+
+#include <facter/facts/collection.hpp>
+#include <facter/util/config.hpp>
+
+#include <hocon/program_options.hpp>
+
+// boost includes are not always warning-clean. Disable warnings that
+// cause problems before including the headers, then re-enable the warnings.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wattributes"
+#include <boost/program_options.hpp>
+#pragma GCC diagnostic pop
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <sstream>
+
+namespace po = boost::program_options;
+
+uint8_t get_default_facts(char **result) {
+    try {
+        po::variables_map vm {};
+        hocon::shared_config hocon_conf;
+        hocon_conf = facter::util::config::load_default_config_file();
```

Comment:
We don't seem to be using any of the settings loaded here. Do we plan to in the future? If not, do we need to load it? (I could be not remembering how it's used somewhere)